### ISSUE TITLE
Upgrading to 4.5.5 Module parse failed: Unexpected token

### ISSUE THREAD

**Issue by eli-taft-quoin** [strapi/strapi#15287]
<!--
Hello 👋 Thank you for submitting an issue.

Before you start, please make sure your issue is understandable and reproducible.
To make your issue readable make sure you use valid Markdown syntax.

https://guides.github.com/features/mastering-markdown/

Please ensure you have also read and understand the contributing guide.

https://github.com/strapi/strapi/blob/main/CONTRIBUTING.md#reporting-an-issue
-->

## Bug report

### Required System information

<!-- Please ensure you are using the Node LTS version (v14 or v16 or v18) -->
<!-- Strapi v3 is not supported unless it is a critical/high security issue -->

- Node.js version:v14.19.3
- NPM version:6.14.17
- Strapi version:4.5.5
- Database:postgresql
- Operating system:Debian 11

### Describe the bug

After upgrading Strapi from 4.1.5 to 4.5.5, I receive many parsing errors while building, indicating "Unexpected token". I will paste two examples of these errors in the "Code snippets" section below.

### Steps to reproduce the behavior

1. Open package.json
2. Update @strapi/plugin-i18n from version 4.1.5 to 4.5.5
3. Update @strapi/plugin-users-permissions from version 4.1.5 to 4.5.5
4. Update @strapi/strapi from version 4.1.5 to 4.5.5
5. Save package.json
6. Run `yarn install`
7. Try to build with `yarn build`

### Expected behavior

Strapi is expected to build successfully.

### Screenshots

N/A

### Code snippets

##### Example errors (there are many like this):

```
  {
    moduleIdentifier: '/home/elimtaft/code/strapi-backend/node_modules/esbuild-loader/dist/index.js??ruleSet[1].rules[2].use!/home/elimtaft/code/strapi-backend/node_modules/@strapi/plugin-users-permissions/admin/src/pages/Providers/index.js',
    moduleName: './node_modules/@strapi/plugin-users-permissions/admin/src/pages/Providers/index.js',
    loc: '170:9',
    message: 'Module parse failed: Unexpected token (170:9)\n' +
      'File was processed with these loaders:\n' +
      ' * ./node_modules/esbuild-loader/dist/index.js\n' +
      'You may need an additional loader to handle the result of these loaders.\n' +
      '|     submitMutation.mutate({ providers: body });\n' +
      '|   });\n' +
      '>   return <Layout>\n' +
      '|     <SettingsPageTitle name={pageTitle} />\n' +
      '|     <Main>',
    moduleId: 4061,
    moduleTrace: [ [Object], [Object], [Object], [Object] ],
    details: undefined,
    stack: 'ModuleParseError: Module parse failed: Unexpected token (170:9)\n' +
      'File was processed with these loaders:\n' +
      ' * ./node_modules/esbuild-loader/dist/index.js\n' +
      'You may need an additional loader to handle the result of these loaders.\n' +
      '|     submitMutation.mutate({ providers: body });\n' +
      '|   });\n' +
      '>   return <Layout>\n' +
      '|     <SettingsPageTitle name={pageTitle} />\n' +
      '|     <Main>\n' +
      '    at handleParseError (/home/elimtaft/code/strapi-backend/node_modules/webpack/lib/NormalModule.js:976:19)\n' +
      '    at /home/elimtaft/code/strapi-backend/node_modules/webpack/lib/NormalModule.js:1095:5\n' +
      '    at processResult (/home/elimtaft/code/strapi-backend/node_modules/webpack/lib/NormalModule.js:800:11)\n' +
      '    at /home/elimtaft/code/react-nextjs-typescript-redux-standard/.subtrees/strapi-backend/node_modules/webpack/lib/NormalModule.js:860:5\n' +
      '    at /home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:406:3\n' +
      '    at iterateNormalLoaders (/home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:232:10)\n' +
      '    at iterateNormalLoaders (/home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:239:10)\n' +
      '    at /home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:254:3\n' +
      '    at context.callback (/home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:124:13)\n' +
      '    at Object.ESBuildLoader (/home/elimtaft/code/strapi-backend/node_modules/esbuild-loader/dist/loader.js:59:9)'
  },
  {
    moduleIdentifier: '/home/elimtaft/code/strapi-backend/node_modules/esbuild-loader/dist/index.js??ruleSet[1].rules[2].use!/home/elimtaft/code/strapi-backend/node_modules/@strapi/plugin-users-permissions/admin/src/pages/Roles/index.js',
    moduleName: './node_modules/@strapi/plugin-users-permissions/admin/src/pages/Roles/index.js',
    loc: '11:9',
    message: 'Module parse failed: Unexpected token (11:9)\n' +
      'File was processed with these loaders:\n' +
      ' * ./node_modules/esbuild-loader/dist/index.js\n' +
      'You may need an additional loader to handle the result of these loaders.\n' +
      '| import ProtectedRolesCreatePage from "./ProtectedCreatePage";\n' +
      '| const Roles = () => {\n' +
      '>   return <CheckPagePermissions permissions={pluginPermissions.accessRoles}><Switch>\n' +
      '|     <Route\n' +
      '|       path={`/settings/${pluginId}/roles/new`}',
    moduleId: 26633,
    moduleTrace: [ [Object], [Object], [Object], [Object] ],
    details: undefined,
    stack: 'ModuleParseError: Module parse failed: Unexpected token (11:9)\n' +
      'File was processed with these loaders:\n' +
      ' * ./node_modules/esbuild-loader/dist/index.js\n' +
      'You may need an additional loader to handle the result of these loaders.\n' +
      '| import ProtectedRolesCreatePage from "./ProtectedCreatePage";\n' +
      '| const Roles = () => {\n' +
      '>   return <CheckPagePermissions permissions={pluginPermissions.accessRoles}><Switch>\n' +
      '|     <Route\n' +
      '|       path={`/settings/${pluginId}/roles/new`}\n' +
      '    at handleParseError (/home/elimtaft/code/strapi-backend/node_modules/webpack/lib/NormalModule.js:976:19)\n' +
      '    at /home/elimtaft/code/strapi-backend/node_modules/webpack/lib/NormalModule.js:1095:5\n' +
      '    at processResult (/home/elimtaft/code/strapi-backend/node_modules/webpack/lib/NormalModule.js:800:11)\n' +
      '    at /home/elimtaft/code/strapi-backend/node_modules/webpack/lib/NormalModule.js:860:5\n' +
      '    at /home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:406:3\n' +
      '    at iterateNormalLoaders (/home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:232:10)\n' +
      '    at iterateNormalLoaders (/home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:239:10)\n' +
      '    at /home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:254:3\n' +
      '    at context.callback (/home/elimtaft/code/strapi-backend/node_modules/loader-runner/lib/LoaderRunner.js:124:13)\n' +
      '    at Object.ESBuildLoader (/home/elimtaft/code/strapi-backend/node_modules/esbuild-loader/dist/loader.js:59:9)'
  }
```

##### package.json

```json
{                                                                                                                                                                                                                                             
  "name": "backend",
  "private": true,
  "version": "0.2.0",
  "description": "A Strapi application, based on Strapi v4",
  "scripts": {
    "develop": "strapi develop",
    "development": "strapi develop",
    "start": "strapi start",
    "production": "strapi start",
    "build": "strapi build",
    "strapi": "strapi"
  },  
  "devDependencies": {}, 
  "dependencies": {
    "@aws-sdk/client-s3": "^3.137.0",
    "@babel/core": "^7.20.7",
    "@strapi/plugin-i18n": "4.5.5",
    "@strapi/plugin-users-permissions": "4.5.5",
    "@strapi/strapi": "4.5.5",
    "@strapi/utils": "^4.5.5",
    "file-saver": "^2.0.5",
    "lodash": "^4.17.21",
    "pg": "^8.7.3",
    "pluralize": "^8.0.0",
    "react-is": "^18.2.0",
    "sqlite3": "5.0.3",
    "strapi-plugin-slugify": "^2.2.1",
    "strapi-provider-email-smtp": "^1.0.3",
    "webpack": "^5.75.0",
    "yup": "^0.32.11"
  },  
  "author": {
    "name": "A Strapi developer"
  },  
  "strapi": {
    "uuid": "9898e2fa-03fd-41a8-9273-63f3d360e692"
  },  
  "engines": {
    "node": ">=14.19.1 <=18.x.x",
    "npm": ">=6.0.0"
  },  
  "license": "MIT"
}
```

### Additional context

Glad to provide more information upon request to help find a solution for this problem.

**Comment by eli-taft-quoin** [comment]
Turns out this is a known issue but the reason is unclear. "it happens when strapi is nested within an existing Node project that also uses babel/webpack". https://forum.strapi.io/t/strapi-develop-command-failed/3875/10. To solve this I had to restructure my project so that Strapi wasn't nested within my NextJS project. Previously my root directory contained the NextJS project that I could build with `yarn build` right from the project root, and the strapi project was in a subdirectory of that. I moved my NextJS into a subdirectory also, so now the project root is basically:

├── .git
├── backend/ (strapi project)
├── frontend/ (NextJS project)
├── README.md

Now Strapi builds successfully. This is ONLY a problem once upgrading Strapi. It was not a problem with the previous version mentioned above.
